### PR TITLE
Update some basic commands in the documentation

### PR DESCRIPTION
**Pull request by weshaggard**
@jaredpar @eerhardt 

These are the basic commands but please let me know if you want to see other things documented. 

skip ci

### CODE REVIEW

File: Documentation/README.md [hunk 1]
```diff
@@ -6,16 +6,25 @@ The following document includes documentation for the build scripts. Over time,
...
+./build.{cmd|sh}
 ```
 
-### Build on Linux or macOS
+### Cleaning
 
 ```console
-./build.sh
+./clean.{cmd|sh}      # Cleans root binary directory and resets all submodules
+./clean.{cmd|sh} -a   # Does a fill git clean on the root repo as well as resets all submodules
```
Comment (eerhardt) [inline Documentation/README.md]: type-o `fill`

File: Documentation/README.md [hunk 2]
```diff
@@ -6,16 +6,25 @@ The following document includes documentation for the build scripts. Over time,
 
 The scripts are supported on Windows, macOS and Linux. The scripts are based on PowerShell on Windows and Bash on macOS and Linux.
 
-### Build on Windows
+### Syncing
+
+This repo uses submodules so remember on fresh clones or when updating/switching branches to run
+
+```
+git submodule update --init
```
Comment (crummel) [inline Documentation/README.md]: NuGet - this should include the `--recursive` flag.

File: Documentation/README.md [hunk 3]
```diff
@@ -6,16 +6,25 @@ The following document includes documentation for the build scripts. Over time,
...
+```
+
+### Building
 
 ```console
-./build.ps1
+./build.{cmd|sh}
```
Comment (crummel) [inline Documentation/README.md]: Do we want to mention that parameters will pass through to MSBuild?

File: Documentation/README.md [hunk 4]
```diff
@@ -6,16 +6,25 @@ The following document includes documentation for the build scripts. Over time,
 
 The scripts are supported on Windows, macOS and Linux. The scripts are based on PowerShell on Windows and Bash on macOS and Linux.
 
-### Build on Windows
+### Syncing
+
+This repo uses submodules so remember on fresh clones or when updating/switching branches to run
```
Comment (dagood) [inline Documentation/README.md]: The doc should probably also mention `git submodule sync [--recursive]` for when `.gitmodules` changes to point to different forks.

File: Documentation/README.md [hunk 5]
```diff
@@ -6,16 +6,25 @@ The following document includes documentation for the build scripts. Over time,
...
-./build.ps1
+./build.{cmd|sh}
 ```
 
-### Build on Linux or macOS
+### Cleaning
 
 ```console
-./build.sh
+./clean.{cmd|sh}      # Cleans root binary directory and resets all submodules
```
Comment (dagood) [inline Documentation/README.md]: I'm not sure how concerned to be about protecting people from accidentally cleaning away their work by only reading the readme, but "resets" is kind of vague. "fully git cleans and hard resets" would be explicit.

For best (imo) user experience, we would have `clean.sh/cmd` by default only run `git clean -xdn`, which lists everything it *would* delete (and do something similar for the `reset --hard`), then prompt. Then we could add `clean.sh/cmd -f` to do it without prompting.

File: Documentation/README.md (hunk 4 above)
Comment (weshaggard) [reply Documentation/README.md]: I'm not sure that will be a common issue so we can keep it out for now.

File: Documentation/README.md (hunk 5 above)
Comment (weshaggard) [reply Documentation/README.md]: I don't really like the two stage process. However if folks end up hitting trouble then we can reconsider.

**Pull request merged**